### PR TITLE
Port smb fixture from test fixture plugin to testcontainer

### PR DESCRIPTION
**Pull request by breskeby**
this removes smb test fixture to use test container instead of relying on our gradle test fixture plugin.

### CODE REVIEW

File: x-pack/qa/third-party/active-directory/src/test/java/org/elasticsearch/xpack/security/authc/ldap/AbstractActiveDirectoryTestCase.java [hunk 1]
```diff
@@ -108,10 +108,6 @@ Settings buildAdSettings(
             .put(getFullSettingKey(realmId, ActiveDirectorySessionFactorySettings.AD_DOMAIN_NAME_SETTING), adDomainName)
             .put(getFullSettingKey(realmName, ActiveDirectorySessionFactorySettings.AD_USER_SEARCH_BASEDN_SETTING), userSearchDN)
             .put(getFullSettingKey(realmName, ActiveDirectorySessionFactorySettings.AD_USER_SEARCH_SCOPE_SETTING), scope)
-            .put(getFullSettingKey(realmName, ActiveDirectorySessionFactorySettings.AD_LDAP_PORT_SETTING), AD_LDAP_PORT)
```
Comment (breskeby) [inline x-pack/qa/third-party/active-directory/src/test/java/org/elasticsearch/xpack/security/authc/ldap/AbstractActiveDirectoryTestCase.java:111]: Tests pass without that configuration. So I removed those, what looks as not required, settings

File: test/fixtures/minio-fixture/build.gradle [hunk 2]
```diff
@@ -11,26 +11,20 @@ apply plugin: 'elasticsearch.cache-test-fixtures'
...
-  implementation "org.slf4j:slf4j-api:${versions.slf4j}"
-  implementation "com.github.docker-java:docker-java-api:${versions.dockerJava}"
-
-  runtimeOnly "com.github.docker-java:docker-java-transport-zerodep:${versions.dockerJava}"
-  runtimeOnly "com.github.docker-java:docker-java-transport:${versions.dockerJava}"
-  runtimeOnly "com.github.docker-java:docker-java-core:${versions.dockerJava}"
-  runtimeOnly "org.apache.commons:commons-compress:${versions.commonsCompress}"
-  runtimeOnly "org.rnorth.duct-tape:duct-tape:${versions.ductTape}"
+  api("org.testcontainers:testcontainers:${versions.testcontainer}") {
+    transitive = false
```
Comment (mark-vieira) [inline test/fixtures/minio-fixture/build.gradle:21]: Hmm. I'm not aware of us doing this anywhere else. There's the larger question of how we want to deal with transitive dependencies going forward but should we just explicitly list them for now?

File: x-pack/test/smb-fixture/src/main/resources/smb/provision/installsmb.sh [hunk 3]
```diff
@@ -8,10 +8,10 @@
 set -ex
 
 VDIR=/fixture
-RESOURCES=$VDIR/src/main/resources
-CERTS_DIR=$RESOURCES/certs
+CERTS_DIR=$VDIR/certs
 SSL_DIR=/var/lib/samba/private/tls
 
+echo "within install smb script"
```
Comment (mark-vieira) [inline x-pack/test/smb-fixture/src/main/resources/smb/provision/installsmb.sh:14]: Is this just debugging?

**Review by mark-vieira — APPROVED** [review]

File: test/fixtures/minio-fixture/build.gradle (hunk 2 above)
Comment (breskeby) [reply test/fixtures/minio-fixture/build.gradle:21]: yeah there is a general clash that we do not disable transitive dependencies for test fixtures but that often clashes with the test class path then resulting  in conflicts etc.

**Pull request merged**